src/Implement/MainManager.cpp

```diff
@@ -0,0 +1,107 @@
+#include "MainManager.h"
+
+
+namespace{
+
+    MainManager* onlyMainManager = nullptr;
+}
+
+void SigHandle(int sig, siginfo_t *info, void *ctx){
+
+    if(sig == SIGUSR1)
+    {
+        mainManager.RemoveByValue(info->si_pid);
+    }
+}
+
+void MainManager::InitInstance(const uint16_t &port_num){
+
+    this.port_num = port_num;
+    HttpListener listen_socket = HttpListener::GetInstance();
+    listen_socket.InitInstance(this.NewRequest, port_num);
+    listen_socket.StartListen();
+}
+
+MainManager* MainManager::GetInstance(){
+
+    if(onlyMainManager == nullptr){
+
+        return new MainManager();
+    }
+    else{
+
+        return onlyMainManager;
+    }
+}
+
+void MainManager::NewRequest(const int &socket_fd){
+
+    pid_t child_pid = fork();
+    //means parent
+    if(child_pid > 0){
+
+        socket_map[socket_fd] = child_pid;
+        struct sigaction act;
+        act.sa_sigaction  = SigHandle;
+        sigemptyset(&act.sa_mask);
+        act.sa_flags = SA_SIGINFO;
+        sigaction(SIGUSR1, &act, NULL);
+    }
+    //means child
+    else if(child_pid == 0){
+
+        RequestHandler request_handler = RequestHandler(socket_fd);
+        request_handler.WaitForMessage();
+    }
+    //fork failed
+    else{
+
+        //TODO: call error
+    }
+    
+}
+
+void MainManager::EndRequest(){
+
+    kill(getppid(), SIGUSR1);
+}
+
+void MainManager::RemoveBySocket(const int &socket_num){
+
+    std::unordered_map<std::string, pid_t>::const_iterator find_process = socket_map.find (std::to_string(socketNum));
+    if(got == tmpInt.end())
+    {
+        //std::cout<<std::to_string(socket_num)<<" not found"<<std::endl;
+        //TODO: call error
+    }
+    else
+    {
+        find_process.erase(std::to_string(socket_num));
```

Comment:
iterator do not have erase function.

Reply:
Done

Reply:
Reviewing about compile error is reasonable. Furthermore, we would usually push for CI so these are automatically found.